### PR TITLE
sort directory context files for determinism

### PR DESCRIPTION
**Pull request by sqs**
This also fixes the Windows unit tests in CI. According to https://github.com/sourcegraph/cody/actions/workflows/ci.yml?query=branch%3Amain, all `main` CI runs including and after https://github.com/sourcegraph/cody/commit/ce072f784a049e44653151961122807638a53224 have been failing on the Windows unit test step.

Also toSorted is not available in Node 18, so we need to use sort still in one other place.

## Test plan

CI

### CODE REVIEW

**Comment by sqs** [comment]
(not urgent) @philipp-spiess do you think we should sort `getWorkspaceFilesContext`'s result for determinism in the tests?

File: agent/src/index.test.ts [hunk 1]
```diff
@@ -789,106 +789,58 @@ describe('Agent', () => {
                 const lastMessage = await client.firstNonEmptyTranscript(id)
                 expect(trimEndOfLine(lastMessage.messages.at(-1)?.text ?? '')).toMatchInlineSnapshot(
                     `
-                  "Based on the shared code context from \`src/example.test.ts\`, the test framework being used is Vitest, which is a modern test runner powered by Vite. It appears to be a test suite for a specific "test block" with multiple test cases.
```
Comment (olafurpg) [inline agent/src/index.test.ts:792]: Reminder to myself to refactor these tests so they become more stable and we stop having to update the snapshots every week. I feel bad for having introduced this abuse of inline snapshot assertions.

File: vscode/src/commands/context/workspace.ts [hunk 2]
```diff
@@ -34,7 +34,9 @@ export async function getWorkspaceFilesContext(
         }, 20000)
 
         try {
-            const results = await workspace.findFiles(globalPattern, excluded, maxResults, token.token)
+            const results = (
+                await workspace.findFiles(globalPattern, excluded, maxResults, token.token)
+            ).sort((a, b) => a.toString().localeCompare(b.toString()))
```
Comment (olafurpg) [inline vscode/src/commands/context/workspace.ts:39]: If the sorting is only needed for testing purposes then you can optionally guard the sort statement behind an if condition. We already do this for symf context.

**Review by olafurpg — APPROVED** [review]
Nice!

File: vscode/src/commands/context/workspace.ts (hunk 2 above)
Comment (sqs) [reply vscode/src/commands/context/workspace.ts:39]: It seems like a good idea to use it for all purposes, not just testing. This ensures deterministic behavior for the same user and across platforms. If it turns out the ordering of findFiles is meaningful, then we should make it consistent across platforms. So, I think this is fine for now.

Comment (olafurpg) [reply vscode/src/commands/context/workspace.ts:39]: Sounds good 👍🏻

**Pull request merged**